dashboard/src/modules/components/DatePickerComponent/index.jsx

```diff
@@ -8,50 +8,57 @@ import {
   Split,
   SplitItem,
   isValidDate,
-  yyyyMMddFormat,
 } from "@patternfly/react-core";
 import React, { useState } from "react";
 import { applyFilter, setFilterKeys } from "actions/datasetListActions";
+import { useDispatch, useSelector } from "react-redux";
 
 import { getTodayMidnightUTCDate } from "utils/dateFunctions";
-import { useDispatch } from "react-redux";
 
 const DatePickerWidget = (props) => {
-  const [from, setFrom] = useState();
-  const [to, setTo] = useState(getTodayMidnightUTCDate());
   const dispatch = useDispatch();
+  const { filter } = useSelector((state) => state.datasetlist);
 
-  const toValidator = (date) =>
-    isValidDate(from) && date >= from
-      ? ""
-      : 'The "to" date must be after the "from" date';
+  const [isEndDateError, setIsEndDateError] = useState(false);
 
   const fromValidator = (date) =>
     date <= getTodayMidnightUTCDate()
       ? ""
       : "The Uploaded date cannot be in the future!";
 
+  const toValidator = (date) =>
+    isValidDate(filter.startDate) && date >= filter.startDate
+      ? ""
+      : 'The "to" date must be after the "from" date';
+
   const onFromChange = (_str, date) => {
-    setFrom(new Date(date));
+    dispatch(setFilterKeys(new Date(date), filter.endDate));
+    if (filter.endDate) {
+      checkEndDate(date, filter.endDate);
+    }
   };
 
   const onToChange = (_str, date) => {
-    if (isValidDate(date)) {
-      setTo(yyyyMMddFormat(date));
+    if (isValidDate(new Date(date))) {
+      dispatch(setFilterKeys(filter.startDate, new Date(date)));
     }
+    checkEndDate(filter.startDate, date);
+  };
```

Comment:
These two functions seem asymmetrical, which seems weird.

- At line 35 we seem sure that `new Date(date)` will produce a valid date, but we don't seem to feel that way at line 42;
- At line 36, we're not sure that `endDate` is valid; at line 45, we're sure that `date` _is_ valid even though we weren't sure three lines earlier.  (Or, at least, we're sure that `checkEndDate()` will do the right thing if the date is invalid...which isn't obvious to me.)

It seems like these functions should look something more like this:
```javascript
  const onFromChange = (_str, date) => {
    dispatch(setFilterKeys(new Date(date), filter.endDate));
    if (filter.endDate) {
      checkEndDate(date, filter.endDate);
    } else {
      setIsEndDateError(true)
    }
  };

  const onToChange = (_str, date) => {
    if (isValidDate(new Date(date))) {
      dispatch(setFilterKeys(filter.startDate, new Date(date)));
      checkEndDate(filter.startDate, date);
    } else {
      setIsEndDateError(true)
    }
  };
```
This way, we always update `isEndDateError`.  Of course, this assumes (I think...) that `onToChange()` is never called before `noFromChange()`....

Also, isn't the `date` parameter equal to `new Date(_str)`?  So, we don't necessarily need to create a new `Date` object from `date`.  (I can see how you might want to do that before placing the reference in the Redux store, but we're doing it twice in `onToChange()` -- once at line 42 and again at line 43 -- I think we can just use `date` at line 42.